Framework/CurveFitting/src/Algorithms/PlotPeakByLogValue.cpp

```diff
@@ -226,160 +235,161 @@ void PlotPeakByLogValue::exec() {
+    double logValue = 0;
+    if (logName.empty() || logName == "axis-1") {
+      API::Axis *axis = data.ws->getAxis(1);
+      if (dynamic_cast<BinEdgeAxis *>(axis)) {
+        double lowerEdge((*axis)(data.i));
+        double upperEdge((*axis)(data.i + 1));
+        logValue = lowerEdge + (upperEdge - lowerEdge) / 2;
+      } else
+        logValue = (*axis)(data.i);
+    } else if (logName != "SourceName") {
+      Kernel::Property *prop = data.ws->run().getLogData(logName);
+      if (!prop) {
+        throw std::invalid_argument("Log value " + logName + " does not exist");
+      }
+      auto *logp = dynamic_cast<TimeSeriesProperty<double> *>(prop);
+      if (!logp) {
+        throw std::runtime_error("Failed to cast " + logName +
+                                 " to TimeSeriesProperty");
+      }
+      logValue = logp->lastValue();
+    }
 
-        g_log.debug() << "Fitting " << data.ws->getName() << " index " << j
-                      << " with \n";
-        g_log.debug() << ifun->asString() << '\n';
-
-        const std::string spectrum_index = std::to_string(j);
-        std::string wsBaseName;
-
-        if (createFitOutput)
-          wsBaseName = wsNames[i].name + "_" + spectrum_index;
-
-        bool histogramFit = getPropertyValue("EvaluationType") == "Histogram";
-        bool ignoreInvalidData = getProperty("IgnoreInvalidData");
-
-        // Fit the function
-        auto fit = this->createChildAlgorithm("Fit");
-        fit->initialize();
-        fit->setPropertyValue("EvaluationType",
-                              getPropertyValue("EvaluationType"));
-        fit->setProperty("Function", ifun);
-        fit->setProperty("InputWorkspace", data.ws);
-        fit->setProperty("WorkspaceIndex", j);
-        fit->setPropertyValue("StartX", getPropertyValue("StartX"));
-        fit->setPropertyValue("EndX", getPropertyValue("EndX"));
-        fit->setProperty("IgnoreInvalidData", ignoreInvalidData);
-        fit->setPropertyValue(
-            "Minimizer", getMinimizerString(wsNames[i].name, spectrum_index));
-        fit->setPropertyValue("CostFunction", getPropertyValue("CostFunction"));
-        fit->setPropertyValue("MaxIterations",
-                              getPropertyValue("MaxIterations"));
-        fit->setPropertyValue("PeakRadius", getPropertyValue("PeakRadius"));
-        fit->setProperty("CalcErrors", true);
-        fit->setProperty("CreateOutput", createFitOutput);
-        if (!histogramFit) {
-          fit->setProperty("OutputCompositeMembers", outputCompositeMembers);
-          fit->setProperty("ConvolveMembers", outputConvolvedMembers);
-          fit->setProperty("Exclude", exclude);
-        }
-        fit->setProperty("Output", wsBaseName);
-        fit->execute();
+    double chi2;
 
-        if (!fit->isExecuted()) {
-          throw std::runtime_error("Fit child algorithm failed: " +
-                                   data.ws->getName());
-        }
+    try {
```

Comment:
The scope of this try catch block is too large and I would argue not needed as we rethrow anyway. Personally I would rather see this entire try / catch block got rid of

If we throw here either our user validation isn't doing it's job or something has gone wrong with the child alg, both of which a user friendly message won't help with